### DISCUSSION TITLE
How to implement deep linking for iOS app on .Net MAUI?

### DISCUSSION THREAD

**Discussion by DavidShochet** [dotnet/maui#14097] [Q&A]
### Discussed in https://github.com/dotnet/maui/discussions/14093

<div type='discussions-op-text'>

<sup>Originally posted by **DavidShochet** March 21, 2023</sup>
I am having a difficulty implementing deep linking on our app. 
Clicking on a link in an email should open the app. As I understand, the execution should come to AppDelegate's OpenUrl() method. The link is for an existing website (https://teams.microsoft.com/meetingOptions/...) that I don't control. 

I was able to easily implement this for Android. Upon clicking on the link, I am asked whether I want to open my app or the MS Teams app, and when I select my app, it just opens. But with iOS I am stuck. I cannot find a clear manual on how this can be done. 

So far, I found that I need to have this in Info.plist:

![](https://i.stack.imgur.com/asTE5.png)

But this is definitely not enough.

I would appreciate your help. </div>

**Comment by moljac** [comment]
for starters

1. Teams custom URL scheme is `msteams` (`msteams://<organization>/<teamName>`)
    From this link seems like replacing `https` with `mstems` should work out of the box
    https://www.eliostruyf.com/devhack-deep-linking-microsoft-teams-android-ios/
    https://rderik.com/blog/creating-app-custom-url-scheme/
     
2. for `https[s]` based schemas I think teams has several options
    1. app `https://teams.microsoft.com/l/app/<your-app-id>`
        https://learn.microsoft.com/en-us/microsoftteams/platform/concepts/build-and-test/deep-link-application?tabs=teamsjs-v2
    2. chat `https://teams.microsoft.com/l/chat/0/0?users=<user1>,<user2>,...&topicName=<chat name>&message=<precanned text>`
        https://learn.microsoft.com/en-us/microsoftteams/platform/concepts/build-and-test/deep-link-teams    
    3. workflow (task) 
        https://learn.microsoft.com/en-us/microsoftteams/platform/concepts/build-and-test/deep-link-workflow?tabs=teamsjs-v2 
        
  Apple calls deep links - universal links
  
  https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html?utm_source=revxblog

Few more links (Xamarin, should be similar for MAUI):

https://learn.microsoft.com/en-us/answers/questions/789113/xamarin-ios-app-responses-on-any-url-using-deep-li

https://learn.microsoft.com/en-us/answers/questions/802339/xamarin-ios-custom-url-scheme-deep-links-work-but  
    
Based on your info I think you should go with universal links for iOS, but before you start try replacing `https` with `msteams` in the url. It MIGHT work.

**Reply by DavidShochet** [reply]
Hi @moljac,

Thank you for your answer. Unfortunately, the links you provided don't address the issue. I am not trying to open or manipulate the Teams app, rather, I need to open my app by clicking on that teams meeting link. And I cannot change the url to a Teams meeting generated by Azure Booking service.
This link https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html?utm_source=revxblog speaks of opening my app by clicking on a link, but it is a link to "my website". I need to handle not a link to my website, but a link to a teams meeting, generated automatically and not controlled by me. 

As I mentioned, I was able to implement it on Android. Android did not require me to create some association file in my website. Is it possible on iOS at all?

This link https://rderik.com/blog/creating-app-custom-url-scheme/ seems to suggest what I did already (see screenshot in my question). But as it did not work for me, it may be either not enough, or I did it incorrectly. Could you please advise? 

Thank you!

**Comment by prabhavmehra** [comment]
@DavidShochet this is how you implement url scheme and not deep linking. They are 2 different things.

**Comment by ToolmakerSteve** [comment]
If you don’t control the website that is referenced in a URL, then you can’t do deep linking into your app.

Sounds like what worked on Android is that the OS recognized your app as supporting video calling.

I don’t know what the equivalent is on iOS. Regardless, the solution won’t be Maui-specific (other than needing to use c# instead of an Apple language). Try asking on StackOverflow.
